dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderFactoryImpl.java

```diff
@@ -1115,18 +1090,24 @@ protected Identifier createIdentifierForFolder(Folder folder, String parentPath)
             sql = sql + " order by " + orderBy;
         }
 
-        dh.setSQLQuery( sql );
-        dh.setFirstResult( offset );
-        dh.setMaxResults( limit );
+        dc.setSQL( sql );
+        dc.setStartRow( offset );
+        dc.setMaxRows( limit );
         if ( identifier.getHostId().equals( Host.SYSTEM_HOST ) ) {
-            dh.setParam( "/" );
-            dh.setParam( identifier.getId() );
+            dc.addParam( "/" );
+            dc.addParam( identifier.getId() );
         } else {
-            dh.setParam( identifier.getURI() + "/" );
-            dh.setParam( identifier.getHostId() );
+            dc.addParam( identifier.getURI() + "/" );
+            dc.addParam( identifier.getHostId() );
         }
 
-        return dh.list();
+        try {
+			return ConvertToPOJOUtil.convertDotConnectMapToPOJO(dc.loadResults(), clazz);
+		}catch(Exception e){
+        	Logger.warn(this, e.getMessage(), e);
+		}
+
+		return new ArrayList<>();

```

Comment:
If the returned collection does not need to be modified you can return Collections.emptyList()

Reply:
As I know, Collections.sort(folderList, Folder::getName:: compareToIgnoreCase) won't work because the method chain is not supported (getName:: compareToIgnoreCase)